src/shogun/features/DenseFeatures.cpp

```diff
@@ -830,20 +651,20 @@ template<> float64_t CDenseFeatures<uint64_t>::dense_dot(
 	return result;
 }
 
-template<> float64_t CDenseFeatures<float32_t>::dense_dot(
+template<> float64_t CDenseFeatures<bool>::dense_dot(
 		int32_t vec_idx1, const float64_t* vec2, int32_t vec2_len) const
 {
 	ASSERT(vec2_len == num_features)
 
 	int32_t vlen;
 	bool vfree;
-	float32_t* vec1 = get_feature_vector(vec_idx1, vlen, vfree);
+	bool* vec1 = get_feature_vector(vec_idx1, vlen, vfree);
 
 	ASSERT(vlen == num_features)
 	float64_t result = 0;
 
 	for (int32_t i = 0; i < num_features; i++)
-		result += vec1[i] * vec2[i];
+		result += vec1[i] ? vec2[i] : 0;
```

Comment:
minor: I wonder whether this additional implementation for `bool` could be avoided via wrapping the memory block with `SGVector<T>(vec1, vlen, false)` and then using `linalg::dot`? (since that might have the dot product for booleans implemented, but not sure actually